### ISSUE TITLE
Flycheck error when open elisp files on Windows

### ISSUE THREAD

**Issue by chuxubank** [hlissner/doom-emacs#4552]
**What did you expect to happen?**
Open elisp file normally with flycheck

**What actually happened?**
Error occur on minibuffer, and flycheck icon get red.

**Additional details:**
- Include a link to your private config
  https://github.com/chuxubank/.doom.d
- Include screenshots/casts of your issue
![image](https://user-images.githubusercontent.com/17001575/105336598-69db1980-5c14-11eb-82d6-10e5a761b222.png)
- If you mention key sequences, include what commands they're bound to (use `SPC
  h k KEY` or `C-h h k KEY` to inspect keys).
- Include any warnings or errors logged to \*Messages\* (use `M-x
  view-echo-area-messages` to see it).
  `Error while checking syntax automatically: (file-error "Spawning child process" "Exec format error")`

<details><pre>
Debugger entered--Lisp error: (file-error "Spawning child process" "Exec format error")
  signal(file-error ("Spawning child process" "Exec format error"))
  (condition-case err (let* ((checker (flycheck-get-checker-for-buffer))) (if checker (flycheck-start-current-syntax-check checker) (flycheck-clear) (flycheck-report-status 'no-checker))) (error (flycheck-report-failed-syntax-check) (signal (car err) (cdr err))))
  (if (flycheck-running-p) nil (run-hooks 'flycheck-before-syntax-check-hook) (flycheck-clear-errors) (flycheck-mark-all-overlays-for-deletion) (condition-case err (let* ((checker (flycheck-get-checker-for-buffer))) (if checker (flycheck-start-current-syntax-check checker) (flycheck-clear) (flycheck-report-status 'no-checker))) (error (flycheck-report-failed-syntax-check) (signal (car err) (cdr err)))))
  (if flycheck-mode (if (flycheck-running-p) nil (run-hooks 'flycheck-before-syntax-check-hook) (flycheck-clear-errors) (flycheck-mark-all-overlays-for-deletion) (condition-case err (let* ((checker (flycheck-get-checker-for-buffer))) (if checker (flycheck-start-current-syntax-check checker) (flycheck-clear) (flycheck-report-status 'no-checker))) (error (flycheck-report-failed-syntax-check) (signal (car err) (cdr err))))) (user-error "Flycheck mode disabled"))
  flycheck-buffer()
  (condition-case err (flycheck-buffer) ((debug error) (message "Error while checking syntax automatically: %S" err) nil))
  (if (or force-deferred (flycheck-must-defer-check)) (flycheck-buffer-deferred) (condition-case err (flycheck-buffer) ((debug error) (message "Error while checking syntax automatically: %S" err) nil)))
  (progn (flycheck--clear-idle-trigger-timer) (setq flycheck--idle-trigger-conditions nil) (if (or force-deferred (flycheck-must-defer-check)) (flycheck-buffer-deferred) (condition-case err (flycheck-buffer) ((debug error) (message "Error while checking syntax automatically: %S" err) nil))))
  (if (and flycheck-mode (if (listp condition) (apply #'flycheck-may-check-automatically condition) (flycheck-may-check-automatically condition))) (progn (flycheck--clear-idle-trigger-timer) (setq flycheck--idle-trigger-conditions nil) (if (or force-deferred (flycheck-must-defer-check)) (flycheck-buffer-deferred) (condition-case err (flycheck-buffer) ((debug error) (message "Error while checking syntax automatically: %S" err) nil)))))
  flycheck-buffer-automatically()
  (progn (flycheck-clean-deferred-check) (flycheck-buffer-automatically))
  (if (flycheck-deferred-check-p) (progn (flycheck-clean-deferred-check) (flycheck-buffer-automatically)))
  (closure (cl-struct-flycheck-syntax-check-tags flycheck-verify-mode-abbrev-table flycheck-verify-mode-syntax-table cl-struct-flycheck-verification-result-tags cl-struct-flycheck-line-cache-tags flycheck-current-errors markdown-fontify-code-blocks-natively markdown-fontify-code-block-default-mode markdown-hide-markup ess-language sh-shell t) nil "Perform the deferred syntax check." (if (flycheck-deferred-check-p) (progn (flycheck-clean-deferred-check) (flycheck-buffer-automatically))))()
  apply((closure (cl-struct-flycheck-syntax-check-tags flycheck-verify-mode-abbrev-table flycheck-verify-mode-syntax-table cl-struct-flycheck-verification-result-tags cl-struct-flycheck-line-cache-tags flycheck-current-errors markdown-fontify-code-blocks-natively markdown-fontify-code-block-default-mode markdown-hide-markup ess-language sh-shell t) nil "Perform the deferred syntax check." (if (flycheck-deferred-check-p) (progn (flycheck-clean-deferred-check) (flycheck-buffer-automatically)))) nil)
  #f(compiled-function (orig-func &rest args) #<bytecode 0x7047eff0d1>)((closure (cl-struct-flycheck-syntax-check-tags flycheck-verify-mode-abbrev-table flycheck-verify-mode-syntax-table cl-struct-flycheck-verification-result-tags cl-struct-flycheck-line-cache-tags flycheck-current-errors markdown-fontify-code-blocks-natively markdown-fontify-code-block-default-mode markdown-hide-markup ess-language sh-shell t) nil "Perform the deferred syntax check." (if (flycheck-deferred-check-p) (progn (flycheck-clean-deferred-check) (flycheck-buffer-automatically)))))
  apply(#f(compiled-function (orig-func &rest args) #<bytecode 0x7047eff0d1>) (closure (cl-struct-flycheck-syntax-check-tags flycheck-verify-mode-abbrev-table flycheck-verify-mode-syntax-table cl-struct-flycheck-verification-result-tags cl-struct-flycheck-line-cache-tags flycheck-current-errors markdown-fontify-code-blocks-natively markdown-fontify-code-block-default-mode markdown-hide-markup ess-language sh-shell t) nil "Perform the deferred syntax check." (if (flycheck-deferred-check-p) (progn (flycheck-clean-deferred-check) (flycheck-buffer-automatically)))) nil)
  flycheck-perform-deferred-syntax-check()
</pre></details>

**Steps to reproduce:**
1. Find a elisp file,
2. Open it.

**System information:**
<details><pre>
```
SYSTEM  type       windows-nt
        config     x86_64-w64-mingw32
        shell      ~/scoop/apps/msys2/current/usr/bin/zsh
        uname      n/a
        path       (C:\Users\$USER\scoop\apps\msys2\current\mingw64\bin C:\Users\$USER\scoop\apps\msys2\current\usr\local\bin C:\Users\$USER\scoop\apps\msys2\current\usr\bin C:\Users\$USER\scoop\apps\msys2\current\usr\bin C:\ProgramData\scoop\shims C:\Windows\system32 C:\Windows C:\Windows\System32\Wbem C:\Windows\System32\WindowsPowerShell\v1.0 C:\Windows\System32\OpenSSH C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common C:\Program Files\Bandizip C:\Users\$USER\scoop\apps\latex\current\texmfs\install\miktex\bin\x64 C:\Users\$USER\scoop\apps\openjdk\current\bin C:\Users\$USER\go\bin C:\Users\$USER\scoop\apps\nodejs\current\bin C:\Users\$USER\scoop\apps\nodejs\current C:\Users\$USER\scoop\shims C:\Users\$USER\doom\bin C:\Users\$USER\AppData\Local\Microsoft\WindowsApps C:\Users\$USER\.dotnet\tools C:\program files\esafenet\cobra docguard client C:\Users\$USER\bin C:\Users\$USER\AppData\Local\Programs\Microsoft VS Code\bin C:\Users\$USER\scoop\apps\msys2\current\usr\bin\site_perl C:\Users\$USER\scoop\apps\msys2\current\usr\bin\vendor_perl C:\Users\$USER\scoop\apps\msys2\current\usr\bin\core_perl ~/scoop/apps/msys2/current/mingw64/libexec/emacs/27.1/x86_64-w64-mingw32/)
EMACS   dir        ~/.emacs.d/
        version    27.1
        build      Nov 20, 2020
        buildopts  --prefix=/mingw64 --build=x86_64-w64-mingw32 --with-modules --without-dbus --without-compress-install 'CFLAGS=-march=x86-64 -mtune=generic -O2 -pipe' CPPFLAGS=-D__USE_MINGW_ANSI_STDIO=1 'LDFLAGS=-pipe -Wl,--dynamicbase,--high-entropy-va,--nxcompat,--default-image-base-high'
        features   XPM JPEG TIFF GIF PNG RSVG SOUND NOTIFY W32NOTIFY ACL GNUTLS LIBXML2 HARFBUZZ ZLIB TOOLKIT_SCROLL_BARS MODULES THREADS JSON PDUMPER GMP
        traits     (gui server-running envvar-file)
DOOM    dir        ~/.doom.d/
        version    2.0.9
        font       #<font-spec nil nil Fira Code nil nil nil nil nil nil nil nil nil nil>
        theme      doom-dracula
        build      HEAD -> fix-msys2-path, chuxubank/fix-msys2-path, chuxubank/develop, develop b4fbafe45 2021-01-21 14:10:14 +0800
        elc-files  0
        modules    (:completion company ivy :ui doom doom-dashboard doom-quit hl-todo modeline nav-flash ophints (popup +defaults) treemacs vc-gutter vi-tilde-fringe window-select workspaces zen :editor (evil +everywhere) file-templates fold (format +onsave) multiple-cursors snippets word-wrap :emacs dired electric undo vc :term eshell :checkers (syntax +childframe) (spell +flyspell) :tools biblio editorconfig (eval +overlay) gist (lookup +docsets +dictionary) lsp magit pass pdf :lang (cc +lsp) data emacs-lisp (json +lsp) (java +lsp) (javascript +lsp) (kotlin +lsp) (latex +latexmk +cdlatex +lsp +fold) markdown (org +dragndrop +hugo +jupyter +pomodoro +present +roam +noter +pretty) plantuml (python +lsp) sh web yaml :app (rss +org) (&user leetcode) :config (default +bindings +smartparens) :input (&user chinese+ +valign) :study (&user review +roam) (&user dict +anki))
        packages   ((lsp-latex) (elfeed-protocol))
        unpin      (n/a)
        elpa       (n/a)
```
</pre></details>

**Comment by chuxubank** [comment]
It works again on my another machine runing Windows, mainly because of the environment.

**Comment by chuxubank** [comment]
The working environment:
```
SYSTEM  type       windows-nt
        config     x86_64-w64-mingw32
        shell      C:/msys64/usr/bin/zsh
        uname      n/a
        path       (C:\msys64\mingw64\bin C:\msys64\usr\local\bin C:\msys64\usr\bin C:\msys64\usr\bin C:\ProgramData\scoop\shims C:\Windows\system32 C:\Windows C:\Windows\System32\Wbem C:\Windows\System32\WindowsPowerShell\v1.0 C:\Windows\System32\OpenSSH C:\Program Files\NVIDIA Corporation\NVIDIA NvDLISR C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common C:\WINDOWS\system32 C:\WINDOWS C:\WINDOWS\System32\Wbem C:\WINDOWS\System32\WindowsPowerShell\v1.0 C:\WINDOWS\System32\OpenSSH C:\Program Files (x86)\NoteBook FanControl C:\Program Files\dotnet C:\Users\$USER\go\bin C:\Users\$USER\scoop\apps\openjdk\current\bin C:\Users\$USER\scoop\apps\latex\current\texmfs\install\miktex\bin\x64 C:\Users\$USER\scoop\apps\bind\current\bin C:\Users\$USER\.emacs.d\bin C:\Users\$USER\AppData\Local\Microsoft\WindowsApps C:\Users\$USER\scoop\shims C:\Users\$USER\AppData\Local\Programs\Microsoft VS Code\bin C:\Users\$USER\scoop\apps\msys2\current\mingw64\bin C:\msys64\usr\bin\site_perl C:\msys64\usr\bin\vendor_perl C:\msys64\usr\bin\core_perl c:/msys64/mingw64/libexec/emacs/27.1/x86_64-w64-mingw32/)
EMACS   dir        ~/.emacs.d/
        version    27.1
        build      Nov 20, 2020
        buildopts  --prefix=/mingw64 --build=x86_64-w64-mingw32 --with-modules --without-dbus --without-compress-install 'CFLAGS=-march=x86-64 -mtune=generic -O2 -pipe' CPPFLAGS=-D__USE_MINGW_ANSI_STDIO=1 'LDFLAGS=-pipe -Wl,--dynamicbase,--high-entropy-va,--nxcompat,--default-image-base-high'
        features   XPM JPEG TIFF GIF PNG RSVG SOUND NOTIFY W32NOTIFY ACL GNUTLS LIBXML2 HARFBUZZ ZLIB TOOLKIT_SCROLL_BARS MODULES THREADS JSON PDUMPER GMP
        traits     (gui server-running envvar-file)
DOOM    dir        ~/.doom.d/
        version    2.0.9
        font       #<font-spec nil nil Fira Code nil nil nil nil nil nil nil nil nil nil>
        theme      doom-dracula
        build      HEAD -> fix-msys2-path, chuxubank/fix-msys2-path, chuxubank/develop b4fbafe45 2021-01-21 14:10:14 +0800
        elc-files  0
        modules    (:completion company ivy :ui doom doom-dashboard doom-quit hl-todo modeline nav-flash ophints (popup +defaults) treemacs vc-gutter vi-tilde-fringe window-select workspaces zen :editor (evil +everywhere) file-templates fold (format +onsave) multiple-cursors snippets word-wrap :emacs dired electric undo vc :term eshell :checkers (syntax +childframe) (spell +flyspell) :tools biblio editorconfig (eval +overlay) gist (lookup +docsets +dictionary) lsp magit pass pdf :lang (cc +lsp) data emacs-lisp (json +lsp) (java +lsp) (javascript +lsp) (kotlin +lsp) (latex +latexmk +cdlatex +lsp +fold) markdown (org +dragndrop +hugo +jupyter +pomodoro +present +roam +noter +pretty) plantuml (python +lsp) sh web yaml :app (rss +org) (&user leetcode) :config (default +bindings +smartparens) :input (&user chinese+ +valign) :study (&user review +roam) (&user dict +anki))
        packages   ((lsp-latex) (elfeed-protocol))
        unpin      (n/a)
        elpa       (n/a)
```
BTY after I reinstall msys2 to it default path, this issue still occur.

**Comment by chuxubank** [comment]
This error occurred randomly, the flyspell also complain process argument about it. (I'll complete it when it occur again.)
Seemed like it occurred only when I find a file from a git repo, after I copy the file to an empty folder, this error disappeared.
